Add a reference that describes all keys and certs used in Corda 5 (#1992)
corda/corda-docs-portal: content/en/platform/corda/5.1/reference/certificates.md

---
title: "Corda Keys and Certificates"
date: '2023-11-28'
version: 'Corda 5.1'
menu:
  corda51:
    identifier: corda51-certificates
    parent: corda51-references
    weight: 4000
section_menu: corda51
---

# Corda Keys and Certificates

This section describes the different types of keys and certificates used by Corda 5. It contains the following:

{{< toc >}}

## Application Keys

> Corda uses the following types of key:

Reviewer: ```suggestion
Corda uses the following types of application key:
```

> Corda uses the following types of keys:

Reviewer 2: ```suggestion
Corda uses the following types of key:
```

| MGM Registration   | Encrypts/decrypts registration requests sent to the MGM by members. The {{< tooltip >}}Group policy{{< /tooltip >}} file includes the public MGM key for members to use when making registration requests. The MGM keeps the private key. Registration uses this key for authentication in the P2P protocol. The MGM registration key is also known as the Corda ECDH key. For more information, see [Configure Key Pairs and Certificates for the MGM]({{< relref "../application-networks/creating/mgm/key-pairs.md#generate-ecdh-key-pair">}}).                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        | 256-bit ECDSA private signature key pair             |

> | Master Wrapping    | Decrypts, through wrapping keys, private keys. This master wrapping key is not available in the database but in higher trust zones (crypto worker and secret managers, for example) that enable decrypting database-encrypted wrapping keys. Encrypted wrapping keys can then decrypt the database-encrypted `VNodePrivateKey`. The crypto processor configuration specifies a set of master wrapping keys as SALT and passphrase parameters for key derivation. This produces a symmetric key which is used for wrapping (encrypting at rest) all corda managed wrapping keys. For more information, see [Key Wrapping]({{< relref "../key-concepts/cluster-admin/_index.md#key-wrapping">}}).                                                                                                                                                                                                                                                                                                                                                           | AES-256 symmetric master key                         |

Reviewer 2: ```suggestion
| Master Wrapping    | Decrypts, through wrapping keys, private keys. This master wrapping key is not available in the database but in higher trust zones (for example, crypto worker and secret managers) that enable decrypting database-encrypted wrapping keys. Encrypted wrapping keys can then decrypt the database-encrypted `VNodePrivateKey`. The crypto processor configuration specifies a set of master wrapping keys as SALT and passphrase parameters for key derivation. This produces a symmetric key which is used for wrapping (encrypting at rest) all Corda-managed wrapping keys. For more information, see [Key Wrapping]({{< relref "../key-concepts/cluster-admin/_index.md#key-wrapping">}}).                                                                                                                                                                                                                                                                                                                                                           | AES-256 symmetric master key                         |
```

| Sensitive Config   | Decrypts sensitive values encrypted at rest (for example, in the database or Kafka). This key is derived from the `Sensitive config SALT and PASSPHRASE` [credential](#credentials).                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      | AES-256 symmetric data encryption key                |

## Infrastructure Keys

> Corda uses the following types of infrastructure keys:

Reviewer 2: ```suggestion
Corda uses the following types of infrastructure key:
```

| REST Gateway TLS | TLS key for REST API/HTTP Gateway connections.          For more information, see [Install the REST Worker Certificate]({{< relref "../deploying-operating/deployment/deploying/_index.md#install-the-rest-worker-certificate" >}}).                                                                                                                                                                                                                                                         | 256-bit ECDSA private authentication key pair   |
| CPK Signing      | Code-signing key for publishing CPKs. The CorDapp Developer uses this key to sign a {{< tooltip >}}CPK{{< /tooltip >}} before bundling it into a CPB. For more information, see [Code Signing]({{< relref "../developing-applications/packaging/code-signing/_index.md">}}).                                                                                                                                                                                                                 | 256-bit X.509 public signature verification key |

> | CPB Signing      | Code-signing key for publishing CPIs. The CorDapp publisher uses this key to sign a {{< tooltip >}}CPB{{< /tooltip >}} before sending it to the Network Operator to use when [creating the CPI]({{< relref "../application-networks/creating/members/cpi.md#import-code-signing-certificates" >}}). For more information, see [Code Signing]({{< relref "../developing-applications/packaging/code-signing/_index.md">}}).                                                                   | 256-bit X.509 public signature verification key |

Reviewer: ```suggestion
| CPB Signing      | Code-signing key for publishing CPIs. The CorDapp publisher uses this key to sign a {{< tooltip >}}CPB{{< /tooltip >}} before sending it to the Network Operator to use when [creating the CPI]({{< relref "../application-networks/creating/members/cpi.md#create-the-cpi-file" >}}). For more information, see [Code Signing]({{< relref "../developing-applications/packaging/code-signing/_index.md">}}).                                                                   | 256-bit X.509 public signature verification key |
```

| CPI Publisher    | Code-signing key for publishing CPIs. Corda validates that uploaded {{< tooltip >}}CPIs{{< /tooltip >}} are signed with a trusted key. The Network Operator usually [imports the signing key to Corda]({{< relref "../application-networks/creating/members/cpi.md#import-code-signing-certificates" >}}). In some cases, such as the Notary CPI, R3 manages the key. For more information, see [Code Signing]({{< relref "../developing-applications/packaging/code-signing/_index.md">}}). | 256-bit X.509 public signature verification key | 256-bit X.509 public signature verification key

> ## PKI assets

Reviewer: ```suggestion
## PKI Assets
```

| Kafka Truststore               | Kafka truststore for server authentication (TLS).                                                                                                                                                      | JKS               |
| Session Keys X.509 Certificate | Session keys certificate validated during session initialization. For more information, see [Session Certificates]({{< relref "../application-networks/creating/optional/session-certificates.md" >}}). | X.509 Certificate |
| E2E Identity Trust Roots       | Trust root of certificates for session initiation keys.                                                                                                                                                | 256-bit PEM       |
| Certificate for Corda releases | Trusted certificates for binary Corda release verification.                                                                                                                                            | X.509 Certificate |

> ## Credentials

Reviewer: Some credentials in this table do not have a type provided. I assumed it's on purpose.

| ---------------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ | ------------------------------------- |
| SSO Config credentials                         | SSO credentials used by the HTTP Gateway process. This is an API ID provided by Azure AD.                                                                                                                                                  | String                                |
| Installation credentials creation              | Installation credentials with enough entropy created in Kubernetes etcd, Helm Chart, or another vault. These credentials are used for key derivation and cluster database secret encryption.                                                 | Passphrase and salt                   |

> | Client Kafka credentials for a specific worker | Kafka credentials (username and password) created at deployment time and stored by default in Kubernetes etcd as secret configuration. This value is passed to worker pods memory at runtime.                                              | High entropy unencrypted string       |

Reviewer: ```suggestion
| Client Kafka credentials for a specific worker | Kafka credentials (user name and password) created at deployment time and stored by default in Kubernetes etcd as secret configuration. This value is passed to worker pods memory at runtime.                                              | High entropy unencrypted string       |
```

| Kubernetes secrets creation via API            | The Cluster Adminstrator creates the secrets in Kubernetes etcd to be used by the Corda helm charts. This is executed optionally when Helm charts overrides are not used as an alternative.                                                |                                       |

> | Kubernetes secrets configured in Helm charts   | The Cluster Adminstrator configures the secrets as overrides in helm charts. This is an alternative to creating Kubernetes secrets via API.                                                                                                                     |                                       |

Reviewer 2: ```suggestion
| Kubernetes Secrets configured in Helm charts   | The Cluster Administrator configures the Secrets as overrides in Helm charts. This is an alternative to creating Kubernetes Secrets via API.                                                                                                                     |                                       |
```

> | Kubernetes Secrets configured in Helm charts   | The Cluster Administrator configures the Secrets as overrides in Helm charts. This is an alternative to creating Kubernetes Secrets via API.                                                                                                                     |                                       |

Reviewer: ```suggestion
| Kubernetes Secrets configured in Helm charts   | The Cluster Administrator configures the secrets as overrides in Helm charts. This is an alternative to creating Kubernetes secrets via API.                                                                                                                     |                                       |
```

Author: Capitalised in the Kubernetes docs so leaving as is.
